js/reference.js

```diff
@@ -973,6 +973,30 @@ var ERMrest = (function(module) {
             }
         },
 
+        /**
+         * create a new reference with the new search
+         * by copying this reference and clears previous search filters
+         * search term can be:
+         * a) A string with no space: single term or regular expression
+         * b) A single term with space using ""
+         * c) use space for conjunction of terms
+         * @param {string} term - search term, undefined to clear search
+         */
+        search: function(term) {
```

Comment:
Should use `verify` to make sure `term` is as described. At least we should check if it's an string or not; it will throw an error in `term.trim()` if it's not an string.